core/src/com/put/urbantraffic/City.java

```diff
@@ -259,8 +259,8 @@ private static void parseGridToClasses(int[][] grid) {
                                     break;
                                 }
                             }
-                            
-                            roads.add(new Road(roads.size(), Math.abs(nodes.get(0).getX() - nodes.get(nodes.size()-1).getX()) + Math.abs(nodes.get(0).getY() - nodes.get(nodes.size()-1).getY()), Arrays.asList(new Lane(laneId, startCrossing, endCrossing, new ArrayList<Direction>()), new Lane(laneId + 1, endCrossing, startCrossing, new ArrayList<Direction>())), nodes));
+
+                            addNewRoad(laneId, nodes, startCrossing, endCrossing);
                             laneId += 2;
```

Comment:
w sumie to chyba tez mozna bylo wydzielic??? nwm nie widze teraz jak ktos wie to niech to doda do brancza